runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -602,12 +632,28 @@ public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
         log.debug("Unregistered channel {}", ctx);
     }
 
+    @Override
+    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+        if (evt.equals(ClientHandshakeEvent.CONNECTED)) {
+            // Handshake successful. Complete the connection future to allow
+            // clients to proceed.
+            connectionFuture.complete(null);
+        } else if (evt.equals(ClientHandshakeEvent.FAILED)) {
+            // Handshake failed. If the current completion future is complete,
+            // create a new one to unset it, causing future requests
+            // to wait.
+            if (connectionFuture.isDone()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)